mode_t missing during compilation

Alpine latest with muslc gives during compilation:
```
thethreefuries.alx nng> make -j2                                   
[  0%] Building C object src/CMakeFiles/nng.dir/platform/posix/posix_epdesc.c.o
[  0%] Building C object src/CMakeFiles/nng_static.dir/platform/posix/posix_epdesc.c.o
In file included from /home/alx/g/nng/src/platform/posix/posix_epdesc.c:14:0:
/home/alx/g/nng/src/platform/posix/posix_aio.h:47:66: error: unknown type name  mode_t'
 extern int  nni_posix_epdesc_set_permissions(nni_posix_epdesc *, mode_t);
                                                                  ^~~~~~
In file included from /home/alx/g/nng/src/platform/posix/posix_epdesc.c:14:0:
/home/alx/g/nng/src/platform/posix/posix_aio.h:47:66: error: unknown type name  mode_t'
 extern int  nni_posix_epdesc_set_permissions(nni_posix_epdesc *, mode_t);
                                                                  ^~~~~~
make[2]: *** [src/CMakeFiles/nng_static.dir/build.make:427: src/CMakeFiles/nng_static.dir/platform/posix/posix_epdesc.c.o] Error 1
make[2]: *** [src/CMakeFiles/nng.dir/build.make:427: src/CMakeFiles/nng.dir/platform/posix/posix_epdesc.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:179: src/CMakeFiles/nng.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:216: src/CMakeFiles/nng_static.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```
adding `#include <sys/types.h>` to the top of `src/platform/posix/posix_aio.h` fixes it.

--- a/src/platform/posix/posix_aio.h
+++ b/src/platform/posix/posix_aio.h
@@ -20,6 +20,8 @@
 #include "core/nng_impl.h"
 #include "posix_pollq.h"
 
+#include <sys/types.h> // needed for mode_t
+
 typedef struct nni_posix_pipedesc nni_posix_pipedesc;
 typedef struct nni_posix_epdesc   nni_posix_epdesc;
 

